GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/GpxFile.java

```diff
@@ -33,54 +36,105 @@ public static File createGpsDataFolder(Context context) {
-            bufferedWriter.flush();
-            bufferedWriter.close();
+            fileWriter.write(writeFileHeader(fileName));
+//            bufferedWriter.flush();
+//            bufferedWriter.close();
             fileWriter.close();
 
         } catch (IOException e) {
-            Log.e(TAG,"Could not create new gpx file.", e);
+            Log.e(TAG, "Could not create new gpx file.", e);
         }
         return gpxFile;
     }
 
     /**
-     * Set the current file name in date format
+     * Get the name for new file
+     *
+     * @param context the context
+     * @return return the name for new file
      */
     public static String getNewFileName(Context context) {
         return getFormattedCurrentTime(context);
     }
 
     /**
-     * Get formatted time
+     * Get formatted system time
+     *
+     * @param context the context
+     * @return return a string of formatted current time
      */
     public static String getFormattedCurrentTime(Context context) {
-        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", context.getResources().getConfiguration().locale);
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'",
+                context.getResources().getConfiguration().locale);
         return sdf.format(System.currentTimeMillis());
     }
 
-    public static void writeFileFooter(File gpxFile, Context context){
+    /**
+     * Create the xml header with version, creator and metadata
+     *
+     * @param formattedTime time of on location changed in format
+     * @return A header string
+     */
+    public static String writeFileHeader(String formattedTime) {
+        StringBuilder header = new StringBuilder();
+        header.append("<?xml version='1.0' encoding='UTF-8' ?>");
+        header.append("<gpx version=\"1.1\" creator=\"GpsDataCapturer " + BuildConfig.VERSION_CODE
+                + "\" ");
+        header.append("xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" ");
+        header.append("xmlns=\"http://www.topografix.com/GPX/1/1\" ");
+        header.append("xsi:schemaLocation=\"http://www.topografix.com/GPX/1/1 ");
+        header.append("http://www.topografix.com/GPX/1/1/gpx.xsd\">");
+        header.append("<metadata><time>").append(formattedTime).append("</time>");
+        header.append("<device>").append(Build.DEVICE).append("</device>");
+        header.append("<id>").append(Build.ID).append("</id>");
+        header.append("<manufacturer>").append(Build.MANUFACTURER).append("</manufacturer>");
+        header.append("<model>").append(Build.MODEL).append("</model></metadata>");
+        header.append("<trk>");
+        header.append("<trkseg>");
+        return header.toString();
+    }
+
+    /**
+     * Write the gpx file footer in xml format
+     *
+     * @param gpxFile the gpxFile
+     * @param context the context
+     */
+    public static void writeFileFooter(File gpxFile, Context context) {
```

Comment:
if follow with instance method idea above: gpxFile becomes a member field, no need to pass as parameter.